presto-password-authenticators/src/main/java/io/prestosql/plugin/password/ldap/LdapConfig.java

```diff
@@ -49,6 +54,31 @@ public LdapConfig setLdapUrl(String url)
         return this;
     }
 
+    public boolean isAllowInsecure()
+    {
+        return allowInsecure;
+    }
+
+    @Config("ldap.allow-insecure")
+    @ConfigDescription("Allow insecure connection to the LDAP server")
+    public LdapConfig setAllowInsecure(boolean allowInsecure)
+    {
+        this.allowInsecure = allowInsecure;
+        return this;
+    }
+
+    @AssertTrue(message = "Connecting to the LDAP server without SSL enabled requires `ldap.allow-insecure=true`")
+    public boolean isUrlConfigurationValid()
+    {
+        if (ldapUrl != null && ldapUrl.startsWith("ldap://")) {
+            if (!allowInsecure) {
+                return false;
+            }
+            log.warn("Passwords will be sent in the clear to the LDAP server. Please consider using SSL to connect.");
```

Comment:
Sorry for the confusion. The logging should be done in  `LdapAuthenticator`, as config classes should be pure configuration data. Also, from a conceptual standpoint, the LDAP authentication is where this matters. Simply creating/validating a config shouldn't cause a warning.

Reply:
So this can be simplified to
```java
return !nullToEmpty(ldapUrl).startsWith("ldap://") || allowInsecure;